src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/CancellationTokenModelBinder.cs

```diff
@@ -16,7 +17,9 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
         {
             if (bindingContext.ModelType == typeof(CancellationToken))
             {
-                var model = bindingContext.OperationBindingContext.HttpContext.RequestAborted;
+                // Force boxing so we can maintain a reference to this token in validation state.
```

Comment:
Not in the existing code they don't. The comment basically says not to remove the cast. I can make it more explicit if necessary.